### ISSUE TITLE
Support and use v6.1 LTS kernel

### ISSUE THREAD

**Issue by stevehipwell** [bottlerocket-os/bottlerocket#2855]
**What I'd like:**
I'd like Bottlerocket to support the `v6.1` LTS kernel and start using it for new versions (once it's been tested). This would align well with AL2023 which it looks like will be using `v6.1` by default.

**Any alternatives you've considered:**
n/a

**Comment by stmcginnis** [comment]
Hey @stevehipwell thanks for this issue. I have heard this being discussed, but I don't believe there has been an issue out there to track it happening. Let's use this issue for that. Thanks!

**Comment by foersleo** [comment]
I have the base enablement for kernel 6.1 [on my fork](https://github.com/foersleo/bottlerocket/tree/kernel-6.1).

Still need to do some testing and we will also need to do some work for ensuring the nvidia drivers are working for the nvidia variants, but I am working on this.

**Comment by foersleo** [comment]
The kernel update in itself seems to work fine. I have successfully done some minimal boot testing and had a look at how much the kernel config changes between 5.15 and 6.1, using a process similar to what we do with `tools/diff-kernel-config`:

```
diff_x86_64:     82 removed, 235 added,   3 changed
diff_aarch64:    92 removed, 226 added,   2 changed
```

Open tasks that I will be working on to get this into the tree:

- [x] Review kernel config changes and see if we need to make any adjustments to what is new
- [x] More elaborate testing
- [x] Create `kmod-6.1-nvidia` package to enable platforms with nvidia drivers

**Comment by foersleo** [comment]
PR for adding the nvidia driver for the new kernel: #3203

**Comment by foersleo** [comment]
With the new kernel packages and accompanying nvidia drivers merged we are set to start using these kernels in the next upcoming variants.

The current plan is to have new variants using 6.1 kernel. The first candidates for that would be k8s-1.28 based variants, aimed to be released with 1.15.0 (https://github.com/bottlerocket-os/bottlerocket/issues/3169) tentatively aiming at August/Early September.

**Comment by stmcginnis** [comment]
Looks like everything is addressed now, so closing out. Feel free to open a new issue if there are any other follow on asks. Thanks!
